src/gui/guiInventoryList.cpp

```diff
@@ -127,26 +128,29 @@ void GUIInventoryList::draw()
 								v2s32(x2 + border, y2)), clipping_rect_ptr);
 		}
 
-		// layer 1
-		if (selected)
-			item.takeItem(m_fs_menu->getSelectedAmount());
-
-		if (!item.empty()) {
-			// Draw item stack
+		// layer 1: draw item stack
+		// (avoid copying the object unless necessary here)
+		if (selected) {
+			ItemStack item2 = item;
+			item2.takeItem(m_fs_menu->getSelectedAmount());
+			if (item2.empty())
+				continue;
+			drawItemStack(driver, m_font, item2, rect, &AbsoluteClippingRect,
+				client, rotation_kind);
+		} else {
 			drawItemStack(driver, m_font, item, rect, &AbsoluteClippingRect,
 					client, rotation_kind);
 		}
 
 		// Add hovering tooltip
 		bool show_tooltip = !item.empty() && hovering && !selected_item;
-		// Make it possible to see item tooltips on touchscreens
-		if (RenderingEngine::getLastPointerType() == PointerType::Touch) {
-			show_tooltip |= hovering && selected && m_fs_menu->getSelectedAmount() != 0;
-		}
+		// make it possible to see item tooltips on touchscreens
+		if (RenderingEngine::getLastPointerType() == PointerType::Touch)
+			show_tooltip |= hovering && selected;
```

Comment:
new bug: https://github.com/minetest/minetest/pull/14029#issuecomment-1825801856, https://github.com/minetest/minetest/commit/db713abea19aff02f3b34d2953ed6f551aee3e42

should add a comment to avoid someone doing the same thing in the future

Reply:
#15607 (I may have been bad at writing this thing)